frameworks/Swift/swift-nio/Sources/swift-nio-tfb-default/main.swift

```diff
@@ -0,0 +1,114 @@
+
+import Foundation
+import NIO
+import NIOHTTP1
+
+struct JSONTestResponse: Encodable {
+	let message = "Hello, World!"
+}
+
+enum Constants {
+	static let httpVersion = HTTPVersion(major: 1, minor: 1)
+	static let serverName = "SwiftNIO"
+
+	static let plainTextResponse: StaticString = "Hello, World!"
+	static let plainTextResponseLength = plainTextResponse.count
+	static let plainTextResponseLengthString = String(plainTextResponseLength)
+
+	static let jsonResponseLength = try! JSONEncoder().encode(JSONTestResponse()).count
+	static let jsonResponseLengthString = String(jsonResponseLength)
+}
+
+private final class HTTPHandler: ChannelInboundHandler {
+
+	public typealias InboundIn = HTTPServerRequestPart
+	public typealias OutboundOut = HTTPServerResponsePart
+
+	let dateFormatter = RFC1123DateFormatter()
+	let jsonEncoder = JSONEncoder()
+
+	func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+		let reqPart = self.unwrapInboundIn(data)
+
+		switch reqPart {
+		case .head(let request):
+			switch request.uri {
+			case "/plaintext":
+				processPlaintext(ctx: ctx)
+			case "/json":
+				processJSON(ctx: ctx)
+			default:
+				_ = ctx.close()
+			}
+		case .body:
+			break
+		case .end:
+			_ = ctx.writeAndFlush(self.wrapOutboundOut(.end(nil)), promise: nil)
```

Comment:
You'll perform better if you just call `write` here, and instead call `flush` in `channelReadComplete`. That will shrink the number of flushes you do, improving performance.